src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs

```diff
@@ -147,6 +147,24 @@ public static CallSettings WithEarlierDeadline(this CallSettings settings, DateT
         }
 
         /// <summary>
+        /// Returns a new <see cref="CallSettings"/> with the given the expiration, merged with the (optional) original settings specified by <paramref name="settings"/>.
+        /// </summary>
+        /// <remarks>
+        /// If <paramref name="settings"/> is non-null and has retry-based timing, the returned settings will have the same timing, but with the specified
+        /// expiration. Otherwise, the returned settings will have a simple expiration.
+        /// </remarks>
+        /// <param name="settings">Existing settings. May be null, meaning there are currently no settings.</param>
+        /// <param name="expiration">Expiration to use in the returned settings, possibly as part of a retry. Must not be null.
+        /// <returns></returns>
+        public static CallSettings WithCallExpiration(this CallSettings settings, Expiration expiration)
```

Comment:
As `settings` can be null - if there weren't any specific settings before, it'll just create a new one.